taf/api/repository.py

```diff
@@ -116,6 +117,42 @@ def create_repository(
         write=True,
         no_commit_warning=True,
     )
+
+    # Create the .git/hooks directory and add the pre-push hook script if it does not exist
+    hooks_dir = Path(auth_repo.path) / ".git" / "hooks"
+    hooks_dir.mkdir(parents=True, exist_ok=True)
+
+    pre_push_script = hooks_dir / "pre-push"
+    if not pre_push_script.exists():
```

Comment:
I haven't tested yet, but let's move this to a separate folder first. We don't currently have similar resources that we copy, so let's create a new folder `resources` in `taf/taf` and then copy from there.